sdk/identity/src/refresh_token.rs

```diff
@@ -23,17 +23,19 @@ pub async fn exchange(
     client_secret: Option<&ClientSecret>,
     refresh_token: &AccessToken,
 ) -> azure_core::Result<RefreshTokenResponse> {
-    let mut encoded = form_urlencoded::Serializer::new(String::new());
-    let encoded = encoded.append_pair("grant_type", "refresh_token");
-    let encoded = encoded.append_pair("client_id", client_id.as_str());
-    // optionally add the client secret
-    let encoded = if let Some(client_secret) = client_secret {
-        encoded.append_pair("client_secret", client_secret.secret())
-    } else {
-        encoded
+    let encoded = {
+        let mut encoded = form_urlencoded::Serializer::new(String::new());
+        let encoded = encoded.append_pair("grant_type", "refresh_token");
+        let encoded = encoded.append_pair("client_id", client_id.as_str());
+        // optionally add the client secret
+        let encoded = if let Some(client_secret) = client_secret {
+            encoded.append_pair("client_secret", client_secret.secret())
+        } else {
+            encoded
+        };
+        let encoded = encoded.append_pair("refresh_token", refresh_token.secret());
+        encoded.finish()
```

Comment:
```suggestion
        let mut encoded = form_urlencoded::Serializer::new(String::new())
            .append_pair("grant_type", "refresh_token");
            .append_pair("client_id", client_id.as_str());
        // optionally add the client secret
        if let Some(client_secret) = client_secret {
            encoded = encoded.append_pair("client_secret", client_secret.secret());
        } 
        encoded.append_pair("refresh_token", refresh_token.secret()).finish()
```

Reply:
I tried that first :) But unfortunately the temporary Serialiser is dropped at the end of the first batch of appends. Couldn't find a neat way to get the conditional append_pair in there. Unless I am missing something obvious?

Reply:
Well, can do:
```
         let mut encoded = form_urlencoded::Serializer::new(String::new());
         let mut encoded = encoded
             .append_pair("grant_type", "refresh_token")
             .append_pair("client_id", client_id.as_str());
         // optionally add the client secret
         if let Some(client_secret) = client_secret {
             encoded = encoded.append_pair("client_secret", client_secret.secret())
         };
         encoded.append_pair("refresh_token", refresh_token.secret())
             .finish()
```

That is maybe a reasonable clean-up?

Reply:
I would probably write it like this:

```rust
let encoded = {
        let mut encoded = &mut form_urlencoded::Serializer::new(String::new());
        encoded = encoded
            .append_pair("grant_type", "refresh_token")
            .append_pair("client_id", client_id.as_str())
            .append_pair("refresh_token", refresh_token.secret());
        // optionally add the client secret
        if let Some(client_secret) = client_secret {
            encoded = encoded.append_pair("client_secret", client_secret.secret())
        }
        encoded.finish()
};
```

Reply:
Sorry - posted mine before I saw yours. Either one would work 😊

Reply:
Went with yours, looks a bit cleaner :)